doc/specifications/proposed/facets.md

```diff
@@ -0,0 +1,303 @@
+- [LocationFacet](/ezsystems/ezpublish-kernel/tree/master/eZ/Publish/API/Repository/Values/Content/Search/Facet/LocationFacet.php)
+- [SectionFacet](/ezsystems/ezpublish-kernel/tree/master/eZ/Publish/API/Repository/Values/Content/Search/Facet/SectionFacet.php)
+- [ContentTypeFacet](/ezsystems/ezpublish-kernel/tree/master/eZ/Publish/API/Repository/Values/Content/Search/Facet/ContentTypeFacet.php)
+- [CriterionFacet](/ezsystems/ezpublish-kernel/tree/master/eZ/Publish/API/Repository/Values/Content/Search/Facet/CriterionFacet.php)
+
+Slider
+
+- [DateRangeFacet](/ezsystems/ezpublish-kernel/tree/master/eZ/Publish/API/Repository/Values/Content/Search/Facet/DateRangeFacet.php)
+- [FieldRangeFacet](/ezsystems/ezpublish-kernel/tree/master/eZ/Publish/API/Repository/Values/Content/Search/Facet/FieldRangeFacet.php)
+
+Display type selection is not a Public API concern but we should enable
+developers to implement a sane selection. Right now `instanceof` checks
+on the facade are probably sufficient. If more facet types are added we
+can still add marker interfaces, like:
+
+- RangeFacet
+- ListFacet
+
+Proposed Changes To The API
+---------------------------
+
+### Unsupported Facets
+
+If a storage engine does not support facets it should just ignore them
+and not throw a exception. The frontend code should expect that less
+facets may be returned then facet builders were provided. Since we
+propose to aggregate the facet builder in the facet (see below) this
+hsould even be less of an issue now.
+
+Given the user flow illustrated above there will be no filters added to
+the facet because there are no facet values for the user to select from.
+
+### Facet Name
+
+I suggest to refactor the `Facet` base class; omit the `$name` property
+and add a `$builder` property referencing the builder used to query the
+facet. This allow to re-use the correct builder in therendering logic
+and maybe reuse the criterion contained in the facet. This also allows
+easy selection of the "active" list entry retrieved from the facet.
+
+If we insist on maintaining BC we can keep the `$name` property in the
+`Facet` class and just deprecate it. We can still fill it or dispatch it
+to the respective property from the builder.
+
+The `$name` püroperty in the `$builder` must be documented that it shall
+contain a user provided name or identifier which is not cared about by
+the API implementation.
+
+**Implementation Note**
+
+To be able to correlate the facet return value from the search backend with the
+correct `FacetBuilder` we should generate a unique ID for each `FacetBuilder`
+instance before executing the query and make the search engine return this ID
+together with the facet return value.
+
+On way to generate such an ID could be as simple as:
+
+    foreach ($query->facets as $facetBuilder) {
+        $facetBuilderMap[md5(serialize($facetBuilder))] = $facetBuilder;
+    }
+
+Instead of `serialize()` we might want to use `spl_object_hash()` or something
```

Comment:
Note: Done in https://github.com/ezsystems/ezplatform-solr-search-engine/commit/d2d4fcb04600e71940598535285366c8e2ce944c using `spl_object_hash()`.